images/egress/router/egress-router.sh

```diff
@@ -130,6 +130,26 @@ function wait_until_killed() {
     wait
 }
 
+function gen_dnsproxy_iptables_rules() {
+    pod_ip=$(ip addr show dev eth0 | grep -Po 'inet \K[\d.]+')
+    if [[ -z "${pod_ip}" ]]; then
+        die "Failed to fetch Pod IP"
+    fi
+      
+    echo -A PREROUTING -p tcp -d "${pod_ip}" -j DNAT --to-destination "${EGRESS_SOURCE}"
```

Comment:
Actually haproxy doesn't need to listen on macvlan interface. It can listen on pod/all interfaces and we don't need this additional iptables forward rule.